doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,397 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
 sort_idx: 20
 ---
 
-## Endpoint profiling ##
-
-
-* [Client-side endpoint profile](#client-side-endpoint-profile)
-  * [Client-side endpoint profile update](#client-side-endpoint-profile-update)
-  * [Client-side endpoint profile example](#the-following-client-side-profile-would-be-compatible-with-our-schema-example)
-* [Server-side endpoint profile](#server-side-endpoint-profile)
-  * [Server-side endpoint profile update](#server-side-endpoint-profile-update)
-  * [Server-side endpoint profile example](#server-side-endpoint-profile-example)
-* [System part of endpoint profile](#system-part-of-endpoint-profile)
-
-The *endpoint profile* is a customizable structured data set that describes specific characteristics of the endpoint. Endpoint profiles are used to classify endpoints into endpoint groups and are comprised of the client-side, server-side and system part. The structure of both client-side and server-side of endpoint profile is defined by application developer using the [Apache Avro schema]() format. Application developer may reuse and share certain data structures using [CTL]().
-
-Client-side structure is used during SDK generation and changes to the client-side structure requires generation of new SDK. Application developer is able to define and change server-side structure of endpoint profile at any time. The structure of the system part is identical across the applications and is used by Kaa internally for its functions. Both client-side and server-side profile schemas are maintained within the corresponding application, with its own version that distinguishes it from the previous schemas. Multiple schema versions and corresponding endpoint profiles created upon those schemas can coexist within a single application.
-
+{% assign root_url = page.url | split: '/'%}
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
+
+* TOC
+{:toc}
+
+The endpoint profile is a customizable structured data set that describes specific characteristics of the endpoint . Endpoint profiles are used to classify endpoints into 
+endpoint groups and are comprised of the client-side, server-side and system part. The structure of both client-side and server-side of endpoint profile is defined by 
+application developer using the [Apache Avro schema](http://avro.apache.org/docs/current/spec.html#schemas) format. Application developer may reuse and share certain data 
+structures using CTL.  
+Client-side structure is used during SDK generation and changes to the client-side structure requires generation of new SDK(for more information use 
+[Endpoint SDK]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs)). Application developer is able to define and 
```

Comment:
Could the above be changed to:

Client-side structure is used during SDK generation. Thus, change to the client-side structure requires re-generation of the SDK (for more details see [Endpoint SDK]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs)).